source/ed/types.cpp

```diff
@@ -59,8 +59,8 @@ void ValidityPattern::add(int duration){
 
 void ValidityPattern::add(boost::gregorian::date start, boost::gregorian::date end, std::bitset<7> active_days){
     for(long i=0; i < (end - start).days(); ++i){
-        boost::gregorian::date current_date = beginning_date + boost::gregorian::days(i);
-        if(active_days[current_date.day_of_week()]){
+        boost::gregorian::date current_date = start + boost::gregorian::days(i);
+        if(active_days[(6 + current_date.day_of_week()) % 7]){
```

Comment:
I don't understand this ?

if it's because the first day is sunday you can use get_weekday(date)

but if it is the case, how did that work before ?